subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/NonIncrementalInputChanges.java

```diff
@@ -53,14 +54,16 @@ public boolean isIncremental() {
     }
 
     private static Stream<FileChange> getAllFileChanges(CurrentFileCollectionFingerprint currentFileCollectionFingerprint) {
-        return currentFileCollectionFingerprint.getFingerprints().keySet().stream().map(RebuildFileChange::new);
+        return currentFileCollectionFingerprint.getFingerprints().entrySet().stream().map(entry -> new RebuildFileChange(entry.getKey(), entry.getValue().getNormalizedPath()));
     }
 
     private static class RebuildFileChange implements FileChange, InputFileDetails {
         private final String path;
+        private final String normalizedPath;
 
-        public RebuildFileChange(String path) {
+        public RebuildFileChange(String path, String normalizedPath) {
```

Comment:
Is this always trivial to calculate? Would it make sense to make it lazy instead?

Reply:
Currently, the normalized path is always trivial to calculate.